addons/mail/static/src/core/common/suggestion_service.js

```diff
@@ -166,11 +166,28 @@ export class SuggestionService {
                 suggestions.push(partner);
             }
         }
+        const specialSuggestions = Object.entries(this.store.specialMentions)
+            .filter(([name, info]) => {
+                return (
+                    info.channel_types.includes(thread.channel_type) &&
+                    cleanedSearchTerm.length >= Math.min(4, name.length) &&
+                    (name.startsWith(cleanedSearchTerm) ||
+                        cleanTerm(info.description.toString()).includes(cleanedSearchTerm))
+                );
+            })
+            .map(([name, info]) => ({
+                isSpecial: true,
+                label: name,
+                type: name,
+                name: info.displayName,
+                description: info.description,
+            }));
```

Comment:
These LOCs would be simpler with `this.store.specialMentions` being a plain array. You'd just need to `specialMentions.filter((val) => {}).map((val) => {})`